sklearn/cluster/optics_.py

```diff
@@ -719,7 +719,7 @@ def _correct_predecessor(reachability_plot, predecessor, ordering, s, e):
     while s < e:
         if reachability_plot[s] > reachability_plot[e]:
             return s, e
-        p_e = predecessor[e]
+        p_e = ordering[predecessor[e]]
```

Comment:
Hmmm. That comes a bit unexpected. I would have expected the predecessor to be an object index, not a plot order index. I'd assume that users will usually expect the predecessor to be in data order, what do you think?

In particular, this is the "externally" used ordering:
https://github.com/scikit-learn/scikit-learn/blob/d83c85d8ab33c375ad57b2bbd5da6f938dab4b08/sklearn/cluster/optics_.py#L163-L164

I'd rather try to get rid of the reindexing here:
https://github.com/scikit-learn/scikit-learn/blob/d83c85d8ab33c375ad57b2bbd5da6f938dab4b08/sklearn/cluster/optics_.py#L624
or at least use some explicit naming, such as `predecessor_plot`. Otherwise, this is a maintenance nightmare, and the next author trying to modify this will be similarly confused and also need 4-5 attempts to get the indexing right...

Reply:
that's great, thanks a lot @adrinjalali @kno10

Reply:
yeah, that's why I renamed the parameters to `predecessor_plot` if the order was the same os `reachability_plot`. We can/should better document this and clarity in our docs later. But I think this is fine for now.